tests/integration/suite/test_node_templates.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+import time
+from .common import random_str
+from .conftest import wait_for
+from rancher import ApiError
+from kubernetes.client import CustomObjectsApi
+
+
+def test_legacy_template_migrate_and_delete(admin_mc, admin_cc,
+                                            remove_resource):
+    """Asserts that any node template not in cattle-global-nt namespace is
+    duplicated into cattle-global-nt, then deleted"""
+    admin_client = admin_mc.client
+    admin_cc_client = admin_cc.client
+    k8s_dynamic_client = CustomObjectsApi(admin_mc.k8s_client)
+
+    ns = admin_cc_client.create_namespace(name="ns-" + random_str(),
+                                          clusterId=admin_cc.cluster.id)
+    remove_resource(ns)
+
+    node_template_name = "nt-" + random_str()
+    body = {
+        "metadata": {
+            "name": node_template_name,
+            "annotations": {
+                "field.cattle.io/creatorId": admin_mc.user.id
+            }
+        },
+        "kind": "NodeTemplate",
+        "apiVersion": "management.cattle.io/v3",
+    }
+
+    node_template = k8s_dynamic_client.create_namespaced_custom_object(
+        "management.cattle.io", "v3", ns.name, 'nodetemplates', body)
+    remove_resource(node_template)
```

Comment:
This doesn't work with k8s client objects, get the obj with our client and pass it into remove_resource OR you can update the fixture to handle k8s object types.